Add missing definitions and declarations for libfm which are
needed by libfm.

Signed-off-by: Dongxiao Xu <dongxiao.xu@intel.com>

diff -ruN libfm-0.1.12-orig/src/gio/fm-app-lookup.h libfm-0.1.12/src/gio/fm-app-lookup.h
--- libfm-0.1.12-orig/src/gio/fm-app-lookup.h	2010-11-17 22:00:48.000000000 +0800
+++ libfm-0.1.12/src/gio/fm-app-lookup.h	2010-11-17 22:01:13.000000000 +0800
@@ -53,6 +53,41 @@
 GType fm_app_lookup_get_type(void);
 void fm_app_lookup_register(GIOModule *module);
 
+#define G_TYPE_DESKTOP_APP_INFO_LOOKUP           (g_desktop_app_info_lookup_get_type ())
+#define G_DESKTOP_APP_INFO_LOOKUP(obj)           (G_TYPE_CHECK_INSTANCE_CAST ((obj), G_TYPE_DESKTOP_APP_INFO_LOOKUP, GDesktopAppInfoLookup))
+#define G_IS_DESKTOP_APP_INFO_LOOKUP(obj)	 (G_TYPE_CHECK_INSTANCE_TYPE ((obj), G_TYPE_DESKTOP_APP_INFO_LOOKUP))
+#define G_DESKTOP_APP_INFO_LOOKUP_GET_IFACE(obj) (G_TYPE_INSTANCE_GET_INTERFACE ((obj), G_TYPE_DESKTOP_APP_INFO_LOOKUP, GDesktopAppInfoLookupIface))
+
+/**
+ * G_DESKTOP_APP_INFO_LOOKUP_EXTENSION_POINT_NAME:
+ *
+ * Extension point for default handler to URI association. See
+ * <link linkend="extending-gio">Extending GIO</link>.
+ */
+#define G_DESKTOP_APP_INFO_LOOKUP_EXTENSION_POINT_NAME "gio-desktop-app-info-lookup"
+
+/**
+ * GDesktopAppInfoLookup:
+ *
+ * Interface that is used by backends to associate default
+ * handlers with URI schemes.
+ */
+typedef struct _GDesktopAppInfoLookup GDesktopAppInfoLookup;
+typedef struct _GDesktopAppInfoLookupIface GDesktopAppInfoLookupIface;
+
+struct _GDesktopAppInfoLookupIface
+{
+  GTypeInterface g_iface;
+
+  GAppInfo * (* get_default_for_uri_scheme) (GDesktopAppInfoLookup *lookup,
+                                             const char            *uri_scheme);
+};
+
+GType     g_desktop_app_info_lookup_get_type                   (void) G_GNUC_CONST;
+
+GAppInfo *g_desktop_app_info_lookup_get_default_for_uri_scheme (GDesktopAppInfoLookup *lookup,
+                                                                const char            *uri_scheme);
+
 G_END_DECLS
 
 #endif /* __FM_APP_LOOKUP_H__ */
